hydra/test_utils/launcher_common_tests.py

```diff
@@ -489,22 +502,25 @@ def test_custom_sweeper_run_workdir(
         expected_dir: str,
         task_launcher_cfg: DictConfig,
         extra_flags: List[str],
+        expected_working_dir: Optional[Path]
```

Comment:
Please explain what expected_working_dir is any why it's needed.
I see it's local to this test which is good, why was it not required before?

Reply:
I've removed params and replaced it with a method instead. (I've realized that adding a params impacts all the existing tests and no easy way to get around that in pytest. )

Also I've renamed this to ```test_app_working_dir```. The test app working dir is the same on local machines as the tests working dir, but is different when tests kick off application on remote machines. which is why this was added.

Reply:
I am having a hard time following the last sentence as it talks about the history if this PR, which is not really relevant right now.

What is this new test testing?

Reply:
sorry for the confusion. The PR is updated now. I didnt rename the test, but just renames the variable's name to ```test_app_working_dir```